### DISCUSSION TITLE
type hole `???`

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8471] [Ideas]
add `???` as a type hole

Questions:
- What is the kind of `???`, maybe kind: `???`
- I guess `???` should always give a type error just by existing

**Comment by mlutze** [comment]
We have Kind.Wild. It's not supposed to make it past kinding, but resilient errors have probably changed that.

**Comment by magnus-madsen** [comment]
Do we have kind error?

**Comment by JonathanStarup** [comment]
next ticket: Kind hole `???`

**Comment by mlutze** [comment]
> Do we have kind error?

Yes

**Comment by magnus-madsen** [comment]
Its clear what it means to compile a program with ??? expressions.

What does it mean to compile a program with ??? type holes?

**Reply by mlutze** [reply]
It means:
1. Type Error
2. We infer the kind if possible
3. We infer the type if possible and put that in the error so we get a nice code action to replace the ??? with the inferred type.

**Reply by magnus-madsen** [reply]
OK; if someone implements that I would accept it.
